plugins/telescope/default.nix

```diff
@@ -5,37 +5,47 @@
   pkgs,
   ...
 }:
-with lib; let
-  cfg = config.plugins.telescope;
-in {
-  imports = [
-    ./file-browser.nix
-    ./frecency.nix
-    ./fzf-native.nix
-    ./fzy-native.nix
-    ./media-files.nix
-    ./ui-select.nix
-    ./undo.nix
-  ];
-
-  # TODO:add support for additional filetypes. This requires autocommands!
-
-  options.plugins.telescope =
-    helpers.neovim-plugin.extraOptionsOptions
-    // {
-      enable = mkEnableOption "telescope.nvim";
-
-      package = helpers.mkPackageOption "telescope.nvim" pkgs.vimPlugins.telescope-nvim;
+with lib;
+# TODO:add support for additional filetypes. This requires autocommands!
+  helpers.neovim-plugin.mkNeovimPlugin config {
+    name = "telescope";
+    originalName = "telescope.nvim";
+    defaultPackage = pkgs.vimPlugins.telescope-nvim;
 
+    maintainers = [maintainers.GaetanLepage];
+
+    extraPackages = [pkgs.bat];
+
+    # TODO introduced 2024-03-24: remove 2024-05-24
+    deprecateExtraOptions = true;
+    optionsRenamedToSettings = ["defaults"];
+
+    imports = [./extensions];
+
+    extraOptions = {
       keymaps = mkOption {
-        type = with types; attrsOf (either str attrs);
+        type = with types;
+          attrsOf
+          (
+            either
+            str
+            (submodule {
+              options = {
+                action = mkOption {
+                  type = types.str;
+                };
+                mode = helpers.keymaps.mkModeOption "n";
+                options = helpers.keymaps.mapConfigOptions;
```

Comment:
Not sure if it's possible for attrsOf submodule, but ideally this should've had a rename deprecation too:

`error: The option `plugins.telescope.keymaps."<leader>bb".desc' does not exist`

Reply:
Yes, this is not really possible unfortunately.
The only alternative is to make a `mkRemovedOptionModule` for the whole `keymaps` option to force the user to re-define it manually.